### PR TITLE
Development

### PR DESCRIPTION
**Pull request by BenMukebo**
### For this React capstone project I build a mobile web application to check a list of metrics (numeric values) that we create making use of React and Redux.

- Select an API that provides numeric data about a topic that I liked
- Build one page with a list of items that could be filtered by some parameters
- Build one page for the item details
- Home page
  - When the page loads, the webapp shows the list of categories that could be filtered out by some parameter.
  - Along with the category name, you will display some numeric values per category that come from the API.
  - When the user clicks (or taps) on a category item, the application navigates to the details page.
- Details page
  - In the details page, the webapp retrieves data from the API to show detailed data in the category.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.
- Create unit tests for pure functions in your Redux code.
- Create integration tests for your application using the React Testing Library.
- Style my components to match the design provided.
- Add search filter bar
- Deploy the project and test for final details.
- Record a video for your project.
- Create a good README and PR description.

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
Hello @BenMukebo 

You've done pretty well so far!
But for a few issues, you are almost done with this. Hang in there!


## Required Changes ♻️
_Please check the comments under the review._


Happy coding, Cheers! 🚀🚀

Do feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, whether it is valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.test.js [hunk 1]
```diff
@@ -1,8 +1,19 @@
...
+    const app = renderer.create(
+      <Provider store={store}>
+        <BrowserRouter>
+          <App />
+        </BrowserRouter>
+      </Provider>,
+    )
+      .toJSON();
+    expect(app).toMatchSnapshot();
+  });
```
Comment (mikenath223) [inline src/App.test.js:18]: - [x] OPTIONAL - Consider adding more unit tests.

File: src/components/css/items.module.css [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  width: 50%;
+  max-width: 400px;
+}
+
+.item-img img {
+  width: 100%;
+  max-width: 400px;
+  opacity: 0.3;
+
+  /* filter: brightness(50%) grayscale(50%) opacity(0.5); */
```
Comment (mikenath223) [inline src/components/css/items.module.css]: - [x] Kindly remove the commented code.

File: src/components/home/Details.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    dispatch(fetchCountry(name));
+  }, []);
+
+  if (loading || !country) {
+    return null;
+  }
+
+  const { All } = country;
+  const list = Object.entries(country).slice();
+  // console.log(list);
```
Comment (mikenath223) [inline src/components/home/Details.js]: - [ ] Kindly remove the commented code.

File: src/components/home/Filter.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+const flterInput = {
+  color: 'black',
+  padding: '0.4rem',
+  border: 'none',
+  outline: 'none',
+  width: '30%',
+};
+
+// eslint-disable-next-line react/prop-types
```
Comment (mikenath223) [inline src/components/home/Filter.js]: - [ ] Kindly remove the eslint disable line and instead fix the issue in the code.

File: src/components/home/Home.js [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    return null;
+  }
+
+  const handleChange = (e) => {
+    setSearchBar(e.target.value);
+    if (!e.target.value) {
+      setLocalItems(items);
+    }
+    const newItems = [...items];
+    // eslint-disable-next-line max-len
```
Comment (mikenath223) [inline src/components/home/Home.js]: - [ ] Kindly remove the eslint disable line and fix the issue in the code.

File: src/index.js [hunk 6]
```diff
@@ -1,17 +1,16 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
+import { Provider } from 'react-redux';
 import './index.css';
 import App from './App';
-import reportWebVitals from './reportWebVitals';
+import store from './redux/configureStore';
+// import './App.css';
```
Comment (mikenath223) [inline src/index.js]: - [ ] Kindly remove the commented code.

File: src/test-utils.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
+/* eslint-disable react/prop-types */
+import React from 'react';
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (mikenath223) [inline src/test-utils.js]: - [ ] Kindly remove the eslint disable line and fix the issue in the code.

File: src/test-utils.js [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    ...renderOptions
+  } = {},
+) {
+  function Wrapper({ children }) {
+    return <Provider store={store}>{children}</Provider>;
+  }
+  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
+}
+
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (mikenath223) [inline src/test-utils.js]: - [ ] Kindly remove the eslint disable line and fix the issue in the code.

File: src/App.test.js (hunk 1 above)
Comment (BenMukebo) [reply src/App.test.js:18]: Thanks so much

File: src/components/css/items.module.css (hunk 2 above)
Comment (BenMukebo) [reply src/components/css/items.module.css]: Resolved

**Review by dasileker — APPROVED** [review]
## **Hi @BenMukebo,** 

## **Congratulations! 🎉 Project Approved ✅**
Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 
 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**